sagemaker-debugger/tensorflow2/tensorflow2_keras_to_estimator/tf2-keras-model-to-estimator.ipynb

```diff
@@ -0,0 +1,407 @@
+{
```

Comment:
<h3>Setting up the SageMaker TensorFlow Estimator </h3><p>Now it's time to setup the SageMaker TensorFlow estimator. We will add the following Debugger-specific parameters to the estimator to enable for debugging the demo training script.</p><ul><li><strong>debugger_hook_config</strong>: This new parameter accepts a local path where you wish your tensors to be written to and also accepts the S3 bucket to store tensors. Debugger will take care of uploading these tensors transparently during execution.</li><li><strong>rules</strong>: This rules parameter will accept a list of rules you want to evaluate the tensors output and training behaviors while the training job is running. There are two types of Debugger rules:</li><li class="ql-indent-1"><strong>Built-in Rules</strong>: These are rules specially curated by the Amazon SageMaker Debugger which you can opt to evaluate your training job.</li><li class="ql-indent-1"><strong>Custom Rules</strong>: You can write your own rules as a Python source file and use them to evaluate your training job. To provide Amazon SageMaker Debugger to evaluate this rule, you have to provide the S3 location of the rule source and the evaluator image. For more information about how to create and use Debugger custom rules, see <a href="https://docs.aws.amazon.com/sagemaker/latest/dg/debugger-custom-rules.html" target="_blank">Create Debugger Custom Rules for Training Job Analysis</a>.</li></ul><h3><br></h3><h3>Using Amazon SageMaker Debugger Rules </h3><p>In this notebook, we will demonstrate how to use SageMaker built-in rules to evaluate the training script. You can find the list of SageMaker rules and the configurations best suited for using them <a href="https://github.com/awslabs/sagemaker-debugger-rulesconfig" target="_blank">here</a>. </p><p>The rules we are using in this notebook are <strong>VanishingGradient</strong> and <strong>LossNotDecreasing</strong>. As the names suggest, the rules will attempt to evaluate if there are vanishing gradients in the tensors captured by the debugging hook during training and also if the loss is not decreasing.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/awslabs/amazon-sagemaker-examples/pull/1298/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Reply:
<p>done in latest</p>